connman/gweb/gweb.c

```diff
@@ -1380,9 +1413,18 @@ static guint do_request(GWeb *web, const char *url,
 		session->address_action = g_timeout_add(0, already_resolved,
 							session);
 	} else {
+		// TODO make to use return value
+		int err = call_dns_route_func(session, true);
+		if (err)
+			debug(session->web, "failed to add DNS routes");
+
 		session->resolv_action = g_resolv_lookup_hostname(web->resolv,
 					host, resolv_result, session);
 		if (session->resolv_action == 0) {
+			err = call_dns_route_func(session, false);
+			if (err)
```

Comment:
if the earlier call_dns_route_func(true) had genuine failure, state is false -> this call_dns_route_func(false) will EALREADY fail -> diagnostic noise?

Reply:
Ah, right, need to handle that EALREADY here as well, thanks.

Reply:
Handled in [a4e3c22](https://github.com/sailfishos/connman/pull/55/commits/a4e3c2202bab7ec34a107eac78bab9c12b6567ff)